core/src/main/java/hudson/model/queue/FutureImpl.java

```diff
@@ -70,17 +78,34 @@ public final Executable waitForStart() throws InterruptedException, ExecutionExc
 
     @Override
     public boolean cancel(boolean mayInterruptIfRunning) {
-        Queue q = Jenkins.get().getQueue();
-        synchronized (q) {
-            synchronized (this) {
-                if(!executors.isEmpty()) {
-                    if(mayInterruptIfRunning)
-                        for (Executor e : executors)
-                            e.interrupt();
-                    return mayInterruptIfRunning;
+        final boolean mayInterrupt = mayInterruptIfRunning;
+        final FutureImpl self = this;
+        
+        //Cancel while holding an exclusive lock on the queue and this future
+        //Note, this is the same order of locking as before, when the queue was also synchronized
+        Callable<Boolean> worker = new Callable<Boolean> () {
+            public Boolean call() {
+                Queue q = Jenkins.getInstance().getQueue();
```

Comment:
```suggestion
                Queue q = Jenkins.get().getQueue();
```
Use non deprecated `get()` instead of `getInstance()`